src/wallet/walletdb.cpp

```diff
@@ -911,11 +1015,20 @@ DBErrors CWalletDB::FindWalletTx(CWallet* pwallet, vector<uint256>& vTxHash, vec
                 uint256 hash;
                 ssKey >> hash;
 
-                CWalletTx wtx;
-                ssValue >> wtx;
+                std::vector<unsigned char> txData(ssValue.begin(), ssValue.end());
+                try {
+                    CWalletTx wtx;
+                    ssValue >> wtx;
+                    vWtx.push_back(wtx);
+                } catch (...) {
+                    // Decode failure likely due to Sapling v4 transaction format change
+                    // between 2.0.0 and 2.0.1. As user is requesting deletion, log the
+                    // transaction entry and then mark it for deletion anyway.
```

Comment:
This is just `FindWalletTx`, so why did we conclude that the user is requesting deletion?

Reply:
This code was added by upstream when adding `-zapwallettxes`, and is only called from `CWalletDB::ZapWalletTx`.

Reply:
Add a comment at the top of the function to that effect. Or rename to `FindWalletTxToZap`.

Reply:
Opened #3572.